src/d3a/d3a_core/export.py

```diff
@@ -169,48 +174,106 @@ def move_root_plot_folder(self):
             shutil.move(os.path.join(old_dir, si), self.plot_dir)
         shutil.rmtree(old_dir)
 
-    def _export_area_with_children(self, area: Area, directory: dir, is_first: bool = False):
+    def _export_spot_markets_stats(self, area: Area, directory: dir, is_first: bool) -> None:
+        """Export bids, offers, trades, statistics csv-files for all spot markets."""
+        self._export_area_stats_csv_file(area, directory, PastDataEnum.SPOT_MARKET, is_first)
+        if not area.children:
+            return
+        self._export_trade_csv_files(past_markets=area.past_markets,
+                                     directory=directory,
+                                     filename=f"{area.slug}-trades",
+                                     is_first=is_first)
+
+        self._export_area_offers_bids_csv_files(past_markets=area.past_markets,
+                                                market_member="offer_history",
+                                                directory=directory,
+                                                filename=f"{area.slug}-offers",
+                                                labels=("slot",) + Offer.csv_fields(),
+                                                is_first=is_first)
+
+        self._export_area_offers_bids_csv_files(past_markets=area.past_markets,
+                                                market_member="bid_history",
+                                                directory=directory,
+                                                filename=f"{area.slug}-bids",
+                                                labels=("slot",) + Bid.csv_fields(),
+                                                is_first=is_first)
+
+    def _export_settlement_markets_stats(self, area: Area, directory: dir, is_first: bool) -> None:
+        """Export bids, offers, trades, statistics csv-files for all settlement markets."""
+        if not ConstSettings.SettlementMarketSettings.ENABLE_SETTLEMENT_MARKETS:
+            return
```

Comment:
Nit: why is this needed? If the settlement markets were not enabled, `area.past_settlement_markets.values()` should be empty, right? If we only take this into consideration, we can avoid adding the dependency from this global setting and make the code a bit more flexible. 

Related to this: if the values of some market slots are not available, should their files (e.g. the trades CSV files) still be generated? If not, we can refactor the `_export_*_csv_files` functions to first check if there is available data. If not, they can just `return` immediately.

Reply:
you are right, there shouldn't be any past settlement markets if the settlement markets are disabled. However, we use the global setting in this module anyway. Is there a performance benefit when checking for whether a dictionary is empty in comparison to checking a global flag?

WRT your second question: CSV files should be generated even if there are no trades or bids or offers in the markets. Which is a pro argument to go with the global flag actually.

In general, this module is really not that flexible. This is really old code and I tried to make it a bit more flexible actually. Due to time contraints I wasn't able to refactor the whole module. This is worth a separate story IMO.

Reply:
> you are right, there shouldn't be any past settlement markets if the settlement markets are disabled. However, we use the global setting in this module anyway. Is there a performance benefit when checking for whether a dictionary is empty in comparison to checking a global flag?
I don't think there's any real performance benefit. There would be a code flexibility benefit though, because we would not need to rely on the global setting to use this function. It's always good to have one less dependency (less error-prone, less refactorings, minor complexity for unit-tests, no teardown issues for unit-tests in this case, etc.).

Reply:
If the CSV files should be generated even without trades, then the flag might still be a good way to proceed anyway, as you said.

Reply:
> If the CSV files should be generated even without trades, then the flag might still be a good way to proceed anyway, as you said.

Thinking from testing perspective, its better to not have an empty CSVs, else from the testing side one need to parse those CSVs too to validate

Reply:
@faizan2590 this is what Hannes answered:
> WRT your second question: CSV files should be generated even if there are no trades or bids or offers in the markets. Which is a pro argument to go with the global flag actually.

I don't know exactly why we want to generate empty CSV files. I would personally be up for not generating them as you said, but if that's a requirement then we should keep them.

Reply:
The case is evident if you check the behavior of the balancing market. Under normal circumstances, when the balancing market is disabled, the csv files for the balancing market are not exported. Give it a try and run the d3a with e.g. default_2a, you will see that no e.g. balancing-offer.csv files will be generated. 

Hannes' case is the same, that he does not want to generate the csv files at all if the settlement market is disabled by configuration. What he also tried to explain and the confusion started, was that there is a case where the settlement markets are enabled, but they do not have any offers / bids / trades due to misconfiguration of the trading agents. In this case, we DO want the csv files to be generated and be empty in order to indicate that settlement markets are enabled but do not have any trades / offers / bids. All these checks can be done only by checking the settlement markets dict, I agree, but I think it is much easier for the reader of the code to see the constant here.